pandas/core/groupby/groupby.py

```diff
@@ -1782,22 +1782,20 @@ def _cython_transform(
 
     @final
     def _transform(self, func, *args, engine=None, engine_kwargs=None, **kwargs):
-        if maybe_use_numba(engine):
-            return self._transform_with_numba(
-                func, *args, engine_kwargs=engine_kwargs, **kwargs
-            )
-
         # optimized transforms
         func = com.get_cython_func(func) or func
 
         if not isinstance(func, str):
-            return self._transform_general(func, *args, **kwargs)
+            return self._transform_general(func, engine, engine_kwargs, *args, **kwargs)
 
         elif func not in base.transform_kernel_allowlist:
             msg = f"'{func}' is not a valid function name for transform(name)"
             raise ValueError(msg)
         elif func in base.cythonized_kernels or func in base.transformation_kernels:
             # cythonized transform or canned "agg+broadcast"
+            if engine is not None:
```

Comment:
Since there are no transformation kernels(e.g. cumsum, shift etc.) for numba yet, this is untested.

Should be OK, though, since this didn't work before anyways.

Reply:
IIRC, it gives something like

``Groupby.method() got an unexpected argument engine``